src/embedding.rs

```diff
@@ -176,8 +177,17 @@ where
         info!("Start saving embeddings.");
 
         let entities_count = self.sparse_matrix_persistor.get_entity_counter();
-        embedding_persistor.put_metadata(entities_count, self.dimension);
+        embedding_persistor
```

Comment:
Should we proceed further when we could not write metadata? Maybe we should we just panic?

Reply:
You are right. If there is a problem then probably further is the same.